applications/CompressiblePotentialFlowApplication/custom_conditions/potential_wall_condition.h

```diff
@@ -0,0 +1,527 @@
+    {
+        Condition::Pointer pNewCondition = Create(NewId, GetGeometry().Create( rThisNodes ), pGetProperties() );
+
+        pNewCondition->SetData(this->GetData());
+        pNewCondition->SetFlags(this->GetFlags());
+
+        return pNewCondition;
+    }
+
+
+    virtual void CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix,
+                                       ProcessInfo& rCurrentProcessInfo) override
+    {
+        VectorType RHS;
+        this->CalculateLocalSystem(rLeftHandSideMatrix,RHS,rCurrentProcessInfo);
+    }
+
+
+
+    /// Calculate wall stress term for all nodes with IS_STRUCTURE != 0.0
+    /**
+      @param rDampingMatrix Left-hand side matrix
+      @param rRightHandSideVector Right-hand side vector
+      @param rCurrentProcessInfo ProcessInfo instance (unused)
+      */
+    virtual void CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                                      VectorType& rRightHandSideVector,
+                                      ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (rLeftHandSideMatrix.size1() != TNumNodes)
+            rLeftHandSideMatrix.resize(TNumNodes,TNumNodes,false);
+        if (rRightHandSideVector.size() != TNumNodes)
+            rRightHandSideVector.resize(TNumNodes,false);
+        rLeftHandSideMatrix.clear();
+
+        array_1d<double,3> An;
+        if(TDim == 2) CalculateNormal2D(An);
+        else CalculateNormal3D(An);
+
+        const array_1d<double,3>& v = this->GetValue(VELOCITY);
+        const double value = -inner_prod(v, An)/static_cast<double>(TNumNodes);
+
+        for(unsigned int i=0; i<TNumNodes; ++i)
+        {
+            rRightHandSideVector[i] = value;
+        }
+
+    }
+
+
+    /// Check that all data required by this condition is available and reasonable
+    virtual int Check(const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        int Check = Condition::Check(rCurrentProcessInfo); // Checks id > 0 and area > 0
+
+        if (Check != 0)
+        {
+            return Check;
+        }
+        else
+        {
+            // Check that all required variables have been registered
+            if(POSITIVE_FACE_PRESSURE.Key() == 0)
+                KRATOS_THROW_ERROR(std::invalid_argument,"POSITIVE_FACE_PRESSURE Key is 0. Check if the application was correctly registered.","");
+            if(NEGATIVE_FACE_PRESSURE.Key() == 0)
+                KRATOS_THROW_ERROR(std::invalid_argument,"NEGATIVE_FACE_PRESSURE Key is 0. Check if the application was correctly registered.","");
+
+            // Checks on nodes
+
+            // Check that the element's nodes contain all required SolutionStepData and Degrees of freedom
+            for(unsigned int i=0; i<this->GetGeometry().size(); ++i)
+            {
+
+                if(this->GetGeometry()[i].SolutionStepsDataHas(POSITIVE_FACE_PRESSURE) == false)
```

Comment:
This is the old KRATOS_THROW_ERROR, it is supposed to be replaced by KRATOS_ERROR